lib/rustler_precompiled.ex

```diff
@@ -802,6 +813,42 @@ defmodule RustlerPrecompiled do
     end)
   end
 
+  defp retry_attempts(options) do
+    retry? = Keyword.get(options, :retry, true)
+
+    if retry? do
+      value = Keyword.get(options, :retry_attempts, 3)
+
+      if not is_integer(value) or value < 1 or value > 15,
+        do: raise("attempts should be between 1 and 15. Got: #{inspect(value)}")
+
+      value
+    else
+      1
+    end
+  end
+
+  defp with_retry(fun, attempts) do
+    task = Task.async(fun)
+
+    Enum.reduce_while(1..attempts, Task.await(task), fn count, partial_result ->
```

Comment:
If you want to allow attempts to be zero, you will need this:

```suggestion
    Enum.reduce_while(1..attempts//1, Task.await(task), fn count, partial_result ->
```